File: DrillingCore.Application/DTOs/DrillInspectionDto.cs
namespace DrillingCore.Application.DTOs
{
    public class DrillInspectionDto
    {
        public int Id { get; set; }
        public string CrewName { get; set; } = default!;
        public string UnitNumber { get; set; } = default!;
        public DateTime DateFilled { get; set; }
        public string? OtherComments { get; set; }
        public int ProjectId { get; set; }

        public List<ChecklistResponseDto> ChecklistResponses { get; set; } = new();
        public List<FormParticipantDto> Participants { get; set; } = new();
        public List<string> PhotoUrls { get; set; } = new();
        public List<FormSignatureDto> Signatures { get; set; } = new();
    }

}

File: DrillingCore.Mobile/drillingcoreamobile/lib/features/forms/models/drill_inspection_dto.dart
import 'package:drillingcoreamobile/features/forms/models/form_signature_dto.dart';
import 'package:drillingcoreamobile/features/forms/models/checklist_response_dto.dart';
import 'package:drillingcoreamobile/features/forms/models/form_participant_dto.dart';


class DrillInspectionDto {
  final int formId;
  final String crewName;
  final String unitNumber;
  final String dateFilled;
  final String? otherComments;
  final int projectId;
  final List<ChecklistResponseDto> checklistResponses;
  final List<FormParticipantDto> participants;
  final List<String> photoUrls;
  final List<FormSignatureDto> signatures;

  DrillInspectionDto({
    required this.formId,
    required this.crewName,
    required this.unitNumber,
    required this.dateFilled,
    required this.otherComments,
    required this.projectId,
    required this.checklistResponses,
    required this.participants,
    required this.photoUrls,
    required this.signatures,
  });

  factory DrillInspectionDto.fromJson(Map<String, dynamic> json) {
    return DrillInspectionDto(
      formId: json['id'],
      crewName: json['crewName'],
      unitNumber: json['unitNumber'],
      dateFilled: json['dateFilled'],
      otherComments: json['otherComments'],
      projectId: json['projectId'],
      checklistResponses: (json['checklistResponses'] as List)
          .map((e) => ChecklistResponseDto.fromJson(e))
          .toList(),
      participants: (json['participants'] as List)
          .map((e) => FormParticipantDto.fromJson(e))
          .toList(),
      photoUrls: List<String>.from(json['photoUrls']),
      signatures: (json['signatures'] as List)
          .map((e) => FormSignatureDto.fromJson(e))
          .toList(),
    );
  }
}